diff --git a/zephyr/program/intelrvp/CMakeLists.txt b/zephyr/program/intelrvp/CMakeLists.txt
index 39a065968d..c4a046348a 100644
--- a/zephyr/program/intelrvp/CMakeLists.txt
+++ b/zephyr/program/intelrvp/CMakeLists.txt
@@ -14,7 +14,6 @@ cros_ec_library_include_directories("${PLATFORM_EC}/driver/tcpm")
 cros_ec_library_include_directories("${PLATFORM_EC}/driver/usb_mux")
 zephyr_library_sources("src/intelrvp.c")
 zephyr_library_sources("src/intel_rvp_board_id.c")
-zephyr_library_sources_ifdef(CONFIG_PLATFORM_EC_DEDICATED_CHARGE_PORT "src/dc_jack.c")
 
 if((DEFINED CONFIG_BOARD_ADLRVP_MCHP) OR (DEFINED CONFIG_BOARD_ADLRVP_NPCX))
   add_subdirectory(adlrvp)
@@ -30,4 +29,5 @@ if((DEFINED CONFIG_BOARD_MTLRVP_M1723) OR (DEFINED CONFIG_BOARD_MTLRVP_MCHP) OR
   zephyr_library_sources_ifdef(CONFIG_PLATFORM_EC_USB_PD_TCPMV2 "src/usb_pd_policy_mecc_1_1.c")
   zephyr_library_sources_ifdef(CONFIG_PLATFORM_EC_USB_PD_TCPMV2 "src/chg_usb_pd_mecc_1_1.c")
   zephyr_library_sources_ifdef(CONFIG_PLATFORM_EC_USB_PD_TCPMV2 "src/chg_usb_tcpc.c")
+  zephyr_library_sources_ifdef(CONFIG_PLATFORM_EC_DEDICATED_CHARGE_PORT "src/dc_jack.c")
 endif()
diff --git a/zephyr/program/intelrvp/include/intelrvp.h b/zephyr/program/intelrvp/include/intelrvp.h
index a240e738d6..1636a5e5cd 100644
--- a/zephyr/program/intelrvp/include/intelrvp.h
+++ b/zephyr/program/intelrvp/include/intelrvp.h
@@ -37,7 +37,7 @@ extern const struct mecc_1_1_tcpc_aic_gpio_config_t mecc_1_1_tcpc_aic_gpios[];
 void board_charging_enable(int port, int enable);
 void board_vbus_enable(int port, int enable);
 void board_set_vbus_source_current_limit(int port, enum tcpc_rp_value rp);
-int board_dc_jack_present(void);
+int board_is_dc_jack_present(void);
 void board_dc_jack_interrupt(enum gpio_signal signal);
 bool is_typec_port(int port);
 #endif /* __INTELRVP_BOARD_H */
diff --git a/zephyr/program/intelrvp/src/chg_usb_tcpc.c b/zephyr/program/intelrvp/src/chg_usb_tcpc.c
index b2866316bd..fbd0732fe5 100644
--- a/zephyr/program/intelrvp/src/chg_usb_tcpc.c
+++ b/zephyr/program/intelrvp/src/chg_usb_tcpc.c
@@ -30,7 +30,7 @@ int board_set_active_charge_port(int port)
 	 * When the Type-C is active port, hardware circuit will
 	 * block DC jack from enabling +VADP_OUT.
 	 */
-	if (port != DEDICATED_CHARGE_PORT && board_dc_jack_present()) {
+	if (port != DEDICATED_CHARGE_PORT && board_is_dc_jack_present()) {
 		LOG_ERR("DC Jack present, Skip enable p%d", port);
 		return EC_ERROR_INVAL;
 	}
diff --git a/zephyr/program/intelrvp/src/dc_jack.c b/zephyr/program/intelrvp/src/dc_jack.c
index 4137644525..b95ecfa4e5 100644
--- a/zephyr/program/intelrvp/src/dc_jack.c
+++ b/zephyr/program/intelrvp/src/dc_jack.c
@@ -17,7 +17,7 @@ bool is_typec_port(int port)
 	return !(port == DEDICATED_CHARGE_PORT || port == CHARGE_PORT_NONE);
 }
 
-int board_dc_jack_present(void)
+int board_is_dc_jack_present(void)
 {
 	return gpio_pin_get_dt(GPIO_DT_FROM_NODELABEL(std_adp_prsnt));
 }
@@ -27,7 +27,7 @@ static void board_dc_jack_handle(void)
 	struct charge_port_info charge_dc_jack;
 
 	/* System is booted from DC Jack */
-	if (board_dc_jack_present()) {
+	if (board_is_dc_jack_present()) {
 		charge_dc_jack.current =
 			(CONFIG_PLATFORM_EC_PD_MAX_POWER_MW * 1000) /
 			DC_JACK_MAX_VOLTAGE_MV;
