diff --git a/zephyr/program/brox/caboc/led_policy.dtsi b/zephyr/program/brox/caboc/led_policy.dtsi
index d507555d24..954ef56f75 100644
--- a/zephyr/program/brox/caboc/led_policy.dtsi
+++ b/zephyr/program/brox/caboc/led_policy.dtsi
@@ -48,20 +48,48 @@
 
                power-state-discharge-batt-low {
                        charge-state = "LED_PWRS_DISCHARGE";
-                       batt-lvl = <BATTERY_LEVEL_EMPTY
-                                   (BATTERY_LEVEL_LOW - 1)>;
+                       batt-lvl = <5 8>;
 
                        battery-led {
                                led-id = "EC_LED_ID_BATTERY_LED";
+                               transition = "LED_TRANSITION_LINEAR";
 
-                               /* Amber 1 sec, off 1 sec */
+                               /* Amber slow flashing */
                                color-0 {
                                        led-color = <&battery_led_amber>;
-                                       period-ms = <1000>;
+                                       period-ms = <0>;
                                };
                                color-1 {
+                                       led-color = <&battery_led_amber>;
+                                       period-ms = <125>;
+                               };
+                               color-2 {
                                        led-color = <&battery_led_off>;
-                                       period-ms = <1000>;
+                                       period-ms = <875>;
+                               };
+                       };
+               };
+
+               power-state-discharge-batt-critical {
+                       charge-state = "LED_PWRS_DISCHARGE";
+                       batt-lvl = <BATTERY_LEVEL_EMPTY 4>;
+
+                       battery-led {
+                               led-id = "EC_LED_ID_BATTERY_LED";
+                               transition = "LED_TRANSITION_LINEAR";
+
+                               /* Amber fast flashing */
+                               color-0 {
+                                       led-color = <&battery_led_amber>;
+                                       period-ms = <0>;
+                               };
+                               color-1 {
+                                       led-color = <&battery_led_amber>;
+                                       period-ms = <125>;
+                               };
+                               color-2 {
+                                       led-color = <&battery_led_off>;
+                                       period-ms = <375>;
                                };
                        };
                };
